opentech/apply/review/views.py

```diff
@@ -21,12 +21,13 @@
 
 class ReviewContextMixin:
     def get_context_data(self, **kwargs):
-        role_reviews = self.object.reviews.filter(
-                author__in=self.object.assigned.with_roles().values('reviewer')
-            )
         # TODO: order role_reviews by role_order
+        role_reviews = self.object.reviews.filter(
```

Comment:
It might be worth just doing two queries, one to get the reviewers and one to get the reviews and then we just merge the data in here. 
```
assigned = self.object.assigned.order_by('role__order').select_related('reviewers')
reviews = self.object.reviews.all().select_related('reviewers')

reviews_block = defaultdict(list)
for reviewer in assigned:
   review = reviews -> magic with reviewer
    if reviewer.is_staff:
         if review:
             key = 'staff_reviewed'
         else:     
              key = 'staff_not_reviewed'
    else:
         if review:
             key = 'other_reviewed'
         else:     
              key = 'other_not_reviewed'

      reviews_block[key].append((reviewer, review))
```

might be a neater way to do it...